test/widgets/message_list_test.dart

```diff
@@ -1758,6 +1759,79 @@ void main() {
       debugNetworkImageHttpClientProvider = null;
     });
 
+    group('User status', () {
+      void checkFindsStatusEmoji(WidgetTester tester, Finder emojiFinder) {
+        final statusEmojiFinder = find.ancestor(of: emojiFinder,
+          matching: find.byType(UserStatusEmoji));
+        check(statusEmojiFinder).findsOne();
+        check(tester.firstWidget<UserStatusEmoji>(statusEmojiFinder)
+          .neverAnimate).isTrue();
+
+        final senderRowFinder = find.ancestor(of: statusEmojiFinder,
+          matching: find.byType(SenderRow));
+        check(senderRowFinder).findsOne();
+      }
+
+      testWidgets('emoji (unicode) & text are set -> emoji is displayed, text is not', (tester) async {
+        final user = eg.user();
+        await setupMessageListPage(tester,
+          users: [user], messages: [eg.streamMessage(sender: user)]);
+        await store.changeUserStatus(user.userId, UserStatusChange(
+          text: OptionSome('Busy'),
+          emoji: OptionSome(StatusEmoji(emojiName: 'working_on_it',
+            emojiCode: '1f6e0', reactionType: ReactionType.unicodeEmoji))));
+        await tester.pump();
+
+        checkFindsStatusEmoji(tester, find.text('\u{1f6e0}'));
+        check(find.descendant(of: find.byType(SenderRow),
+          matching: find.text('Busy'))).findsNothing();
+      });
+
+      testWidgets('emoji (image) & text are set -> emoji is displayed, text is not', (tester) async {
+        prepareBoringImageHttpClient();
+
+        final user = eg.user();
+        await setupMessageListPage(tester,
+          users: [user], messages: [eg.streamMessage(sender: user)]);
+        await store.changeUserStatus(user.userId, UserStatusChange(
+          text: OptionSome('Coding'),
+          emoji: OptionSome(StatusEmoji(emojiName: 'zulip',
+            emojiCode: 'zulip', reactionType: ReactionType.zulipExtraEmoji))));
+        await tester.pumpAndSettle();
```

Comment:
Best to avoid pumpAndSettle: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-using-pumpandsettle

Can this be two pumps? Better to be explicit that way.